### ISSUE TITLE
train_and_evaluate hang forever without message

### ISSUE THREAD

**Issue by Continue7777** [tensorflow/tensorflow#28071]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow):no
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):Linux 3.10.0-693.11.6.el7.x86_64
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary):binary
- TensorFlow version (use command below):1.6.0
- Python version:2.7
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version:CUDA Version 9.0.176
- GPU model and memory:P100 12.0G

**Describe the current behavior**
when LD_LIBARARY_PATH don't include the libhdfs.so,and we set estimator RunConfig-model_dir to a hdfs path, it will hang at tf.train_and_evaluate forever .I think it can't operate tf.gfile.MkDir() ,but it gives no info or any message.

**Describe the expected behavior**
I expect that it will catch the exception and give the info about loss xx.so.

**Code to reproduce the issue**
Provide a reproducible test case that is the bare minimum necessary to generate the problem.

classifier = tf.estimator.Estimator(
        model_fn=text_cnn_fn,
        params={
            'feature_columns': feature_columns,
            'vocab_size': FLAGS.hash_size,
            'embedding_size': FLAGS.embedding_size,
            'filter_sizes': filter_sizes,
            'num_filters': FLAGS.num_filters,
            'sequence_length': FLAGS.sequence_length,
            'n_classes': FLAGS.num_classes,
            'learning_rate': FLAGS.learning_rate,
        },
        config=tf.estimator.RunConfig(model_dir="hdfs://default/tmp")
    )
...data process
 tf.estimator.train_and_evaluate(classifier, train_spec, eval_spec)

**Other info / logs**
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached.

**Comment by ymodak** [comment]
In order to expedite the trouble-shooting process, please provide a minimal code snippet to reproduce the issue reported here. Thanks!

**Comment by ymodak** [comment]
Automatically closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!
